src/veh_interact.cpp

```diff
@@ -371,6 +378,15 @@ void veh_interact::cache_tool_availability()
                 crafting_inv.has_components( "wheel_bicycle", 1 ) ||
                 crafting_inv.has_components( "wheel_motorbike", 1 ) ||
                 crafting_inv.has_components( "wheel_small", 1 );
+
+    int qual_jack = ceil( double( veh->total_mass() * 1000 ) / TOOL_LIFT_FACTOR );
+    has_jack = g->u.has_quality( "JACK", qual_jack ) ||
+               map_selector( g->u.pos(), PICKUP_RANGE ).has_quality( "JACK", qual_jack ) ||
+               vehicle_selector( g->u.pos(), 1, *veh ).has_quality( "JACK", qual_jack );
```

Comment:
Cranes can't lift themselves so by definition you're stood adjacent to another vehicle your working on. The only requirement is that that crane is adjacent to you - your other crafting tools are found from the usual `PICKUP_RANGE`.